### ISSUE TITLE
Suspended playback on Edge/IE aswell as MSE error on Samsung browser

### ISSUE THREAD

**Issue by CurtisConway** [google/shaka-player#2041]
Open up Edge and IE and load any of the following manifests into the demo player:

https://player.vimeo.com/external/345746172.m3u8?s=38bceb69f832c994e0950d5af157be5d9257ad73

https://player.vimeo.com/external/344197872.m3u8?s=0ae01be5ccea6163bc62299ddfcd6b4a3c2b8198

https://player.vimeo.com/external/342374546.m3u8?s=3f0a9afe7b419d63b689fff2e1b984410bcfa9d5

You'll notice that the playback is suspended and having trouble progressing for the first few seconds of the video.

If you seek to any other part in the video the playback continues as expected. If you seek back to 0 seconds the playback will be suspended again.

**My Question:**
Is this a symptom of Edge and IE performance? Or is this a symptom with the Manifest itself? Perhaps with the redirect on the initial playlist request? Is there maybe a gap or something in the first few seconds that the library has trouble dealing with?

For what it's worth over the past couple years I've test-driven pretty much every Javascript ABR player and library under the sun, and Shaka was the first not to give me this behavior in more popular browsers. Which leads me to believe the issue lies in the manifests themselves.

I'd like to get another opinion before I discount it like that though. I've contacted Vimeo about this issue and they've refused to acknowledge it. Unfortunately, I don't know enough at this point to challenge their stance on the matter.

**Comment by CurtisConway** [comment]
I'm really hoping someone with some expertise can give me a hand with this issue.

I've done a bit deeper of a dive this time into the issue, but I'm still unsure what's actually causing it.

Here's a repo with a code sample, mirroring a very simple example of my configuration.
https://curtisconway.github.io/shaka-player-demos/
https://github.com/CurtisConway/shaka-player-demos/blob/master/scripts.js

If we load this sample on EDGE or IE you'll notice that the top video by default just starts at 3 seconds (despite it being told to start at 0). If you try to seek the video to any point before that it just stalls out and doesn't play (no errors are actually observed though).

If we load the sample on Samsung Browser (which is as far as I'm aware, Chromium-based). We can observe the top video failing with a CHUNK_DEMUXER_ERROR_APPEND_FAILED error passed through the Media Source Extensions.

In all 3 cases, the player that was set to start at 20 seconds was able to load the manifest at that point properly without issue. You can see that the startup code is virtually exactly the same with only the start time parameter being different.

So this makes me think the issue is 100% in the manifest, but I'm still not convinced. Taking a look at one of the playlists inside the master manifest. It looks like this:

```
#EXTM3U
#EXT-X-VERSION:3
#EXT-X-MEDIA-SEQUENCE:0
#EXT-X-TARGETDURATION:6
#EXT-X-PLAYLIST-TYPE:VOD
#EXTINF:6.066667
chop/segment-1.ts
#EXTINF:6.066667
chop/segment-2.ts
#EXTINF:6.033333
chop/segment-3.ts
#EXTINF:6.066667
chop/segment-4.ts
#EXTINF:6.066667
chop/segment-5.ts
#EXTINF:6.066667
chop/segment-6.ts
#EXTINF:6.166667
chop/segment-7.ts
#EXT-X-ENDLIST
```

I did a quick read of the HLS specification defined at https://tools.ietf.org/html/draft-pantos-hls-rfc8216bis-04#section-4.4.2.1

The first thing I noticed was the documentation for the `EXT-X-TARGETDURATION` tag.

> The EXT-X-TARGETDURATION tag specifies the maximum Media Segment duration.  The EXTINF duration of each Media Segment in the Playlist file, when rounded to the nearest integer, **MUST be less than or equal to the target duration; longer segments can trigger playback stalls or other errors.**  It applies to the entire Playlist file.

As you can see in the playlist above, the `EXT-X-TARGETDURATION` tag is defined at exactly 6 seconds. And every segment has a duration of slightly more than 6 seconds. So this makes me believe the issue lies in the manifest itself, and I should contact Vimeo about this. But the fact that the playlist works fine on other browsers tells me it's not _that_ big of an issue.

One last thing I did was run the manifest through Apple's HLS Media Stream Validator tool. The validator tool didn't come back in any warnings/errors about the durations but it did give me some _MUST FIX_ issues regarding the bitrates. So now I'm totally confused again.

We can observe the output from that below:

```
[/external/344197872.m3u8] Started root playlist download

[/1565655409-0xc3d1134067aff33014cad9a3f1811c6b9a8081b8/344197872/video/1379909061/playlist.m3u8] Started media playlist download

[/1565655409-0xc3d1134067aff33014cad9a3f1811c6b9a8081b8/344197872/video/1379909053/playlist.m3u8] Started media playlist download

[/1565655409-0xc3d1134067aff33014cad9a3f1811c6b9a8081b8/344197872/video/1379909056/playlist.m3u8] Started media playlist download

[/1565655409-0xc3d1134067aff33014cad9a3f1811c6b9a8081b8/344197872/video/1379909057/playlist.m3u8] Started media playlist download

[/1565655409-0xc3d1134067aff33014cad9a3f1811c6b9a8081b8/344197872/video/1379909058/playlist.m3u8] Started media playlist download

[/1565655409-0xc3d1134067aff33014cad9a3f1811c6b9a8081b8/344197872/video/1379909056/playlist.m3u8] All media files delivered and have end tag, stopping

[/1565655409-0xc3d1134067aff33014cad9a3f1811c6b9a8081b8/344197872/video/1379909053/playlist.m3u8] All media files delivered and have end tag, stopping

[/1565655409-0xc3d1134067aff33014cad9a3f1811c6b9a8081b8/344197872/video/1379909058/playlist.m3u8] All media files delivered and have end tag, stopping

[/1565655409-0xc3d1134067aff33014cad9a3f1811c6b9a8081b8/344197872/video/1379909057/playlist.m3u8] All media files delivered and have end tag, stopping

[/1565655409-0xc3d1134067aff33014cad9a3f1811c6b9a8081b8/344197872/video/1379909061/playlist.m3u8] All media files delivered and have end tag, stopping



--------------------------------------------------------------------------------

https://skyfire.vimeocdn.com/1565655409-0xc3d1134067aff33014cad9a3f1811c6b9a8081b8/344197872/video/1379909061/playlist.m3u8

--------------------------------------------------------------------------------

Processed 7 out of 7 segments

Average segment duration: 6.076191

Total segment bitrates (all discontinuities): average: 3837.43 kb/s, max: 6290.52 kb/s

Playlist max bitrate: 6112.001000 kb/s





Discontinuity: sequence: 0, parsed segment count: 7 of 7, duration: 42.533 sec, average: 3837.43 kb/s, max: 6290.52 kb/s

Track ID: 1

Video Codec: avc1

Video profile: High

Video level: 4

Video resolution: 1920x1080

Video average IDR interval: 2.828889, Standard deviation: 0.514798

Video frame rate: 30.000

Track ID: 2

Audio Codec: AAC-LC

Audio sample rate: 48000 Hz

Audio channels: 2

Audio channel layout: Stereo (L R)



--------------------------------------------------------------------------------

https://skyfire.vimeocdn.com/1565655409-0xc3d1134067aff33014cad9a3f1811c6b9a8081b8/344197872/video/1379909053/playlist.m3u8

--------------------------------------------------------------------------------

Processed 7 out of 7 segments

Average segment duration: 6.076191

Total segment bitrates (all discontinuities): average: 656.64 kb/s, max: 736.30 kb/s

Playlist max bitrate: 689.415000 kb/s





Discontinuity: sequence: 0, parsed segment count: 7 of 7, duration: 42.533 sec, average: 656.64 kb/s, max: 736.30 kb/s

Track ID: 1

Video Codec: avc1

Video profile: High

Video level: 3

Video resolution: 640x360

Video average IDR interval: 3.030952, Standard deviation: 0.008585

Video frame rate: 30.000

Track ID: 2

Audio Codec: AAC-LC

Audio sample rate: 48000 Hz

Audio channels: 2

Audio channel layout: Stereo (L R)



--------------------------------------------------------------------------------

https://skyfire.vimeocdn.com/1565655409-0xc3d1134067aff33014cad9a3f1811c6b9a8081b8/344197872/video/1379909056/playlist.m3u8

--------------------------------------------------------------------------------

Processed 7 out of 7 segments

Average segment duration: 6.076191

Total segment bitrates (all discontinuities): average: 374.50 kb/s, max: 426.16 kb/s

Playlist max bitrate: 384.531000 kb/s





Discontinuity: sequence: 0, parsed segment count: 7 of 7, duration: 42.533 sec, average: 374.50 kb/s, max: 426.16 kb/s

Track ID: 1

Video Codec: avc1

Video profile: High

Video level: 2.1

Video resolution: 426x240

Video average IDR interval: 3.030952, Standard deviation: 0.008585

Video frame rate: 30.000

Track ID: 2

Audio Codec: AAC-HEv2

Audio sample rate: 48000 Hz

Audio channels: 2

Audio channel layout: Stereo (L R)



--------------------------------------------------------------------------------

https://skyfire.vimeocdn.com/1565655409-0xc3d1134067aff33014cad9a3f1811c6b9a8081b8/344197872/video/1379909057/playlist.m3u8

--------------------------------------------------------------------------------

Processed 7 out of 7 segments

Average segment duration: 6.076191

Total segment bitrates (all discontinuities): average: 1438.32 kb/s, max: 1863.56 kb/s

Playlist max bitrate: 1780.075000 kb/s





Discontinuity: sequence: 0, parsed segment count: 7 of 7, duration: 42.533 sec, average: 1438.32 kb/s, max: 1863.56 kb/s

Track ID: 1

Video Codec: avc1

Video profile: High

Video level: 3.1

Video resolution: 960x540

Video average IDR interval: 3.030952, Standard deviation: 0.008585

Video frame rate: 30.000

Track ID: 2

Audio Codec: AAC-LC

Audio sample rate: 48000 Hz

Audio channels: 2

Audio channel layout: Stereo (L R)



--------------------------------------------------------------------------------

https://skyfire.vimeocdn.com/1565655409-0xc3d1134067aff33014cad9a3f1811c6b9a8081b8/344197872/video/1379909058/playlist.m3u8

--------------------------------------------------------------------------------

Processed 7 out of 7 segments

Average segment duration: 6.076191

Total segment bitrates (all discontinuities): average: 2254.66 kb/s, max: 3095.43 kb/s

Playlist max bitrate: 2984.853000 kb/s





Discontinuity: sequence: 0, parsed segment count: 7 of 7, duration: 42.533 sec, average: 2254.66 kb/s, max: 3095.43 kb/s

Track ID: 1

Video Codec: avc1

Video profile: High

Video level: 3.2

Video resolution: 1280x720

Video average IDR interval: 2.828889, Standard deviation: 0.514798

Video frame rate: 30.000

Track ID: 2

Audio Codec: AAC-LC

Audio sample rate: 48000 Hz

Audio channels: 2

Audio channel layout: Stereo (L R)



--------------------------------------------------------------------------------

MUST fix issues

--------------------------------------------------------------------------------



Error: Measured peak bitrate compared to master playlist declared value exceeds error tolerance

--> Detail:  Measured: 426.16 kb/s, Master playlist: 384.53 kb/s, Error: 10.83%

--> Source:  https://player.vimeo.com/external/344197872.m3u8?s=0ae01be5ccea6163bc62299ddfcd6b4a3c2b8198

--> Compare: https://skyfire.vimeocdn.com/1565655409-0xc3d1134067aff33014cad9a3f1811c6b9a8081b8/344197872/video/1379909056/playlist.m3u8



Error: Measured average bitrate compared to master playlist declared value exceeds error tolerance

--> Detail:  Measured: 374.50 kb/s, Master playlist: 333.00 kb/s, Error: 12.46%

--> Source:  https://player.vimeo.com/external/344197872.m3u8?s=0ae01be5ccea6163bc62299ddfcd6b4a3c2b8198

--> Compare: https://skyfire.vimeocdn.com/1565655409-0xc3d1134067aff33014cad9a3f1811c6b9a8081b8/344197872/video/1379909056/playlist.m3u8
```

**Comment by joeyparrish** [comment]
@CurtisConway, I apologize that someone has not responded sooner.  We've been a bit behind on issues lately.

Should we take it that this is no longer an issue with a newer version of Shaka Player?  (Also, what version were you using?)  Or are you just frustrated with us and giving up?  Again, please accept my apologies.

**Comment by CurtisConway** [comment]
@joeyparrish - I should be the one apologizing. I thought I submitted a comment along with closing it. I guess I forgot to actually press "Comment".

The issue still exists but I don't think it has anything to do with Shaka directly. Which is why I posted it as a question instead of a bug. I'm pretty sure the issue is with the source files at this point, even my post above is a lot of evidence for exactly that.

Since you guys are extremely busy I don't want to bog you down with my personal blockers.

I'm going to continue doing my own research and possibly reopen this in a more relevant forum. If someone personally has some time to take a look into it I'd love if they could send me an email with their thoughts: `curtis@drumeo.com`. My current knowledge of ABR is pretty rudimentary and I'd love the opportunity to learn some more.
